src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -239,6 +239,31 @@ public static void Main()
         }
 
         [Fact]
+        public void EmitRefAssemblyWhenTargetExe()
+        {
+            CSharpCompilation comp = CreateCompilationWithMscorlib(@"
+public class C
+{
+    internal static void Main()
+    {
+        System.Console.WriteLine(""hello"");
+    }
+}
+", options: TestOptions.DebugExe);
+
+            using (var output = new MemoryStream())
+            using (var metadataOutput = new MemoryStream())
+            {
+                // Previously, this would crash when trying to get the entry point for the ref assembly
+                // (but the Main method is not emitted in the ref assembly...)
+                EmitResult emitResult = comp.Emit(output, metadataPEStream: metadataOutput,
+                    options: new EmitOptions(includePrivateMembers: false));
+                Assert.True(emitResult.Success);
```

Comment:
>Assert.True(emitResult.Success); [](start = 16, length = 32)

Consider also verifying metadata, i.e. that the entry point is not set as expected. The fact that there is no crash doesn't mean that expected metadata emitted.